File: backend/app/models/plant.py
from sqlalchemy import Column, Integer, String, Text, JSON, Float, DateTime
from sqlalchemy.ext.declarative import declarative_base
from datetime import datetime

Base = declarative_base()

class MedicinalPlant(Base):
    __tablename__ = "medicinal_plants"

    id = Column(Integer, primary_key=True, index=True)

    # Names
    local_name = Column(String, index=True, nullable=False)
    scientific_name = Column(String, index=True, nullable=False)
    common_names = Column(JSON)  # Store multiple common names

    # Classification
    family = Column(String)
    genus = Column(String)
    species = Column(String)

    # Physical Description
    description = Column(Text)
    habitat = Column(Text)
    distribution = Column(Text)

    # Phytochemical Data
    active_compounds = Column(JSON)  # List of phytochemicals
    chemical_composition = Column(JSON)  # Detailed composition data

    # Therapeutic Properties
    medicinal_uses = Column(JSON)  # List of therapeutic uses
    preparation_methods = Column(JSON)  # How to prepare
    dosage_info = Column(Text)
    contraindications = Column(Text)
    side_effects = Column(Text)

    # Additional Information
    traditional_knowledge = Column(Text)
    research_references = Column(JSON)  # Academic references

    # Image Data
    image_urls = Column(JSON)  # Multiple plant images

    # Metadata
    confidence_score = Column(Float)
    verified = Column(Integer, default=0)  # 0: unverified, 1: verified
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)

File: frontend/src/types/plant.ts
export interface MedicinalPlant {
  id: number;

  // Names
  local_name: string;
  scientific_name: string;
  common_names: string[];

  // Classification
  family: string;
  genus: string;
  species: string;

  // Physical Description
  description: string;
  habitat: string;
  distribution: string;

  // Phytochemical Data
  active_compounds: string[];
  chemical_composition: Record<string, any>;

  // Therapeutic Properties
  medicinal_uses: string[];
  preparation_methods: string[];
  dosage_info: string;
  contraindications: string;
  side_effects: string;

  // Additional Information
  traditional_knowledge: string;
  research_references: ResearchReference[];

  // Image Data
  image_urls: string[];

  // Metadata
  confidence_score: number;
  verified: boolean;
  created_at: string;
  updated_at: string;
}

export interface ResearchReference {
  title: string;
  authors?: string[];
  journal?: string;
  year?: number;
  url?: string;
}

export interface IdentificationResult {
  plant: MedicinalPlant;
  confidence: number;
  matches: MedicinalPlant[];
}

export interface IdentificationHistory {
  id: number;
  plant_id: number;
  plant: MedicinalPlant;
  image_url: string;
  confidence: number;
  identified_at: string;
}

export interface PlantSearchParams {
  query?: string;
  family?: string;
  medicinal_use?: string;
  limit?: number;
  offset?: number;
}

export interface PlantSearchResponse {
  plants: MedicinalPlant[];
  total: number;
  offset: number;
  limit: number;
}
